src/allmydata/webish.py

```diff
@@ -217,36 +217,50 @@ def censor(queryargs: bytes) -> bytes:
     return urlencode(result, safe="[]").encode("ascii")
 
 
+def anonymous_tempfile(tempdir: bytes) -> BinaryIO:
```

Comment:
The type signature is wrong, should be `Callable[[], BinaryIO]`. And maybe function should have name indicating this creates a tempfile factory, not a tempfile?